packages/@aws-cdk/aws-route53-targets/lib/bucket-website-target.ts

```diff
@@ -0,0 +1,32 @@
+import route53 = require('@aws-cdk/aws-route53');
+import s3 = require('@aws-cdk/aws-s3');
+import {Stack, Token} from '@aws-cdk/core';
+import regionInfo = require('@aws-cdk/region-info');
+
+/**
+ * Use a S3 as an alias record target
+ */
+export class BucketWebsiteTarget implements route53.IAliasRecordTarget {
+    constructor(private readonly bucket: s3.Bucket) {
+    }
+
+    public bind(_record: route53.IRecordSet): route53.AliasRecordTargetConfig {
+        const {region} = Stack.of(this.bucket.stack);
+
+        // FIXME
+        if (Token.isUnresolved(region)) {
+            throw new Error(`Unresolved region token: ${region}`);
```

Comment:
Provide a clearer error message. Something like: 

    Cannot use an S3 record alias in region-agnostic stacks. You must specify a specific region when you define the stack (see https://docs.aws.amazon.com/cdk/latest/guide/environments.html)

(We need a URL shortener...)

Reply:
I thought I was missing something to try and detect the region on the fly. Something like this wouldn't work?

https://github.com/aws/aws-cdk/blob/4a0272db5eb9aae5f440bb0813fdbad6514b60c3/packages/%40aws-cdk/aws-iam/lib/principals.ts#L333-L355

Reply:
If my previous comment is wrong, done

Reply:
No, this won't help. For region-agnostics stacks, `stack.region` will always resolve to `{ Ref: "AWS::Region"}`. This is intentional in order to enforce that there is constructs do not make region-related assumptions.